config/v1alpha1/types_cluster_monitoring.go

```diff
@@ -101,3 +106,232 @@ const (
+	// userMode is required.
+	// Allowed values are Selectable and None
+	// Default value is None
+	// +kubebuilder:validation:Enum="";Selectable;None
+	// +optional
+	UserMode UserAlertManagerMode `json:"userMode"`
+	// logLevel defines the verbosity of logs emitted by Alertmanager.
+	// This field allows users to control the amount and severity of logs generated, which can be useful
+	// for debugging issues or reducing noise in production environments.
+	// Allowed values are Error, Warn, Info, Debug, and omitted.
+	// When set to Error,  only errors will be logged.
+	// When set to Warn, both warnings and errors will be logged.
+	// When set to Info, general information, warnings, and errors will all be logged.
+	// When set to Debug, detailed debugging information will be logged.
+	// When omitted, this means no opinion and the platform is left to choose a default that is subject to change over time.
+	// Currently, the default is Info.
+	// +optional
+	LogLevel LogLevel `json:"logLevel"`
+	// nodeSelector is the node selector applied to network diagnostics components
+	// nodeSelector is optional.
+	//
+	// When omitted, this means the user has no opinion and the platform is left
+	// to choose reasonable defaults. These defaults are subject to change over time.
+	// The current default is `kubernetes.io/os: linux` so that Pods can be scheduled onto any available node.
+	// +optional
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+	// resources defines the compute resource requests and limits for the Alertmanager container.
+	// This includes CPU, memory and HugePages constraints to help control scheduling and resource usage.
+	// When not specified, defaults are used by the platform. Requests cannot exceed limits.
+	// This field is optional.
+	// More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
+	// This is a simplified API that maps to Kubernetes ResourceRequirements.
+	// +optional
+	Resources *AlertmanagerContainerResources `json:"resources,omitempty"`
+	// secrets Defines a list of secrets that need to be mounted into the Alertmanager.
+	// The secrets must reside within the same namespace as the Alertmanager object.
+	// They will be added as volumes named secret-<secret-name> and mounted at
+	// /etc/alertmanager/secrets/<secret-name> within the 'alertmanager' container of
+	// the Alertmanager Pods.
+	// This field is optional.
+	// Maximum length for this list is 10
+	// +optional
+	// +kubebuilder:validation:MaxItems=10
+	Secrets []SecretName `json:"secrets,omitempty"`
+	// tolerations is a list of tolerations applied to network diagnostics components
+	// tolerations is optional.
+	//
+	// When omitted, this means the user has no opinion and the platform is left
+	// to choose reasonable defaults. These defaults are subject to change over time.
+	// The current default is `- operator: "Exists"` which means that all taints are tolerated.
```

Comment:
Is that safe? Not even an API question, but, tolerating all taints is generally not something we would do for control plane components. There are many valid taints (uninitialized for the CCM, network not ready) that I would expect this pod not to tolerate